### DISCUSSION TITLE
Get rid of Shape, IndexMode/EditMode and Extent

### DISCUSSION THREAD

**Discussion by jany-tenaj** [DotSpatial/DotSpatial#1419] [Ideas]
I'm thinking of getting rid of Shape, IndexMode/EditMode and Extent.

As far as I have read, Extent was added, so we didn't have to reference NetTopologySuite in every Project. And something about Envelope being heavier that Extent. If I have a look at Extent and Envelope source code I can't follow the second argument because the code seems to be almost the same. 
Including NetToplogySuite is quite simple, so this shouldn't be a reason to have two classes that basically do the same.

What I see is that we have to convert between Envelope and Extent whenever we switch between Geometries and display related things. Furthermore there seem to be errors inside Extent methods that were fixed in Envelope at one point but not in Extent. 

Based on the fact, that we have not much time I don't see the benefit of maintaining two different code versions that basically do the same.

The same arguments can be used for Shapes. Basically they are a duplicate of Geometry. They look a little different inside, but do basically the same.

The functions for Geometries are maintained by the NTS people but there are some functions for Shapes that contain errors.

IndexMode/EditMode is basically used to switch between Features with Geometries and Shapes and wouldn't be needed any longer if we remove Shapes. 

The fact that one can switch between both more often than not causes side effects as seen in #1417 as well as #801.

**Comment by sindizzy** [comment]
I think simplicity is the best approach. I've always been confused as to why we have to use Extent and Envelope.
